exchanges/poloniex/poloniex_types.go

```diff
@@ -66,6 +66,27 @@ type TradeHistory struct {
 	Total         float64 `json:"total,string"`
 }
 
+// OrderStatus holds order status data
+type OrderStatus struct {
+	Result  map[string]interface{} `json:"result"`
```

Comment:
`Result  map[string]interface{} `json:"result"` can be changed to:
```go
Result  map[string]wsStatus `json:"result"`

// include new type
// wsStatus defines order status details
type wsStatus struct {
	Pair           string  `json:"currencyPair"`
	Rate           float64 `json:"rate,string"`
	Amount         float64 `json:"amount,string"`
	Total          float64 `json:"total,string"`
	StartingAmount float64 `json:"startingAmount,string"`
	Type           string  `json:"type"`
	Status         string  `json:"status"`
	Date           string  `json:"date"`
	Fee            float64 `json:"fee,string"`
}
```

This should cut down on type handling.